src/modules/samples/rust/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+use libc::{c_int, c_char ,c_void};//, c_void
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::ptr::NonNull;
+
+mod sample;
+
+// Constants and Method headers were generated by bindgen
+// (https://github.com/rust-lang/rust-bindgen)
+pub const MMI_OK: u32 = 0;
+pub type MMI_HANDLE = *mut ::std::os::raw::c_void;
+pub type MMI_JSON_STRING = *mut ::std::os::raw::c_char;
+
+// MSFT change: Added the #[no_mangle] annotation and defined the methods
+#[no_mangle]
+pub extern "C" fn MmiGetInfo(
+        clientName: *const ::std::os::raw::c_char,
+        payload: *mut MMI_JSON_STRING,
+        payloadSizeBytes: *mut ::std::os::raw::c_int,
+    ) -> ::std::os::raw::c_int {
+    let c_str: &CStr = unsafe { CStr::from_ptr(clientName) };
+    let str_slice: &str = c_str.to_str().unwrap();
+    println!("{}",str_slice);
+    // Dereferencing a pointer
+    unsafe{*payloadSizeBytes = 5};
+    // Converting a string to a c_char pointer
+    unsafe{let temppayload = CString::new("newpayload").unwrap();
+    *payload= CString::into_raw(temppayload);}
+    // Dummy return
+    return 0;
+}
+
+#[no_mangle]
+pub extern "C" fn MmiOpen(
+        clientName: *const ::std::os::raw::c_char,
+        maxPayloadSizeBytes: ::std::os::raw::c_uint,
+    ) -> MMI_HANDLE {
+    // let sample = sample::Sample({})
+    let c_str: &CStr = unsafe { CStr::from_ptr(clientName) };
+    let str_slice: &str = c_str.to_str().unwrap();
+    println!("Client_name: {}",str_slice);
+    println!("{}", maxPayloadSizeBytes);
+    let mut vec = vec![1, 2, 3];
+    let result =  vec.as_mut_ptr() as *mut _ as *mut c_void;
+    return result;
+}
+
+#[no_mangle]
+pub extern "C" fn MmiClose(clientSession: MMI_HANDLE){
+    return;
+}
+
+#[no_mangle]
+pub extern "C" fn MmiSet(
+        clientSession: MMI_HANDLE,
+        componentName: *const ::std::os::raw::c_char,
+        objectName: *const ::std::os::raw::c_char,
+        payload: MMI_JSON_STRING,
+        payloadSizeBytes: ::std::os::raw::c_int,
+    ) -> ::std::os::raw::c_int {
+    return 0;
+}
+
+#[no_mangle]
+pub extern "C" fn MmiGet(
+        clientSession: MMI_HANDLE,
+        componentName: *const ::std::os::raw::c_char,
+        objectName: *const ::std::os::raw::c_char,
+        payload: *mut MMI_JSON_STRING,
+        payloadSizeBytes: *mut ::std::os::raw::c_int,
+    ) -> ::std::os::raw::c_int {
+    return 0;
+}
+
+#[no_mangle]
+pub extern "C" fn MmiFree(payload: MMI_JSON_STRING) {
+    return;
+}
+
+// #[derive(Clone, Copy)]
```

Comment:
For all the commented-out code, please remove until you are read to merge it in. It will likely change as you continue working so it may be best to leave it out until it is ready.